src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
+    final boolean tooFewTerritories = numTerritories <= 1;
+    final boolean tooFewPoints = points.length <= 2;
+    final int xOffset = view.getXOffset();
+    final int yOffset = view.getYOffset();
+    final double scale = view.getScale();
+    final int jointsize = 10;
+    if (tooFewTerritories || tooFewPoints) {
+      drawLineWithTranslate(graphics, new Line2D.Float(routeDescription.getStart(), routeDescription.getEnd()), xOffset,
+          yOffset, scale);
+      graphics.fillOval((int) (((routeDescription.getEnd().x - xOffset) - jointsize / 2) * scale),
+          (int) (((routeDescription.getEnd().y - yOffset) - jointsize / 2) * scale), jointsize, jointsize);
+    } else {
+      drawCurvedPath(graphics, points, view);
+      drawMoveLength(graphics, routeDescription, points, xOffset, yOffset, view, mapData.scrollWrapX(),
+          mapData.scrollWrapY(), numTerritories, movementLeftForCurrentUnits);
+    }
+    drawJoints(graphics, points, xOffset, yOffset, jointsize, scale);
+    final Image cursorImage = routeDescription.getCursorImage();
+    if (cursorImage != null) {
+      graphics.drawImage(cursorImage,
+          (int) (((routeDescription.getEnd().x - xOffset) - (cursorImage.getWidth(null) / 2)) * scale),
+          (int) (((routeDescription.getEnd().y - yOffset) - (cursorImage.getHeight(null) / 2)) * scale), null);
     }
   }
 
-  private static void drawWithTranslate(final Graphics2D graphics, final Shape shape, final double translateX,
-      final double translateY) {
-    if (shape instanceof Ellipse2D.Double) {
-      Ellipse2D.Double elipse = (Ellipse2D.Double) shape;
-      elipse = new Ellipse2D.Double(elipse.x + translateX, elipse.y + translateY, elipse.width, elipse.height);
-      graphics.draw(elipse);
+  private static void drawJoints(Graphics2D graphics, Point[] points, int xOffset, int yOffset, int jointsize,
+      double scale) {
+    for (Point p : points) {
+      graphics.fillOval((int) (((p.x - xOffset) - jointsize / 2) * scale),
+          (int) (((p.y - yOffset) - jointsize / 2) * scale), jointsize, jointsize);
     }
-    if (shape instanceof Polygon) {
-      ((Polygon) shape).translate((int) translateX, (int) translateY);
-      graphics.fill(shape);
-      ((Polygon) shape).translate((int) -translateX, (int) -translateY);
+
+  }
+
+  protected static double[] getIndex(Point[] points) {
+    final double[] index = new double[points.length];
+    if(index.length > 0){
+      index[0] = 0;
     }
-    if (shape instanceof Line2D) {
-      final Line2D line = (Line2D) shape;
-      final Point2D p1 = new Point2D.Double(line.getP1().getX() + translateX, line.getP1().getY() + translateY);
-      final Point2D p2 = new Point2D.Double(line.getP2().getX() + translateX, line.getP2().getY() + translateY);
-      graphics.draw(new Line2D.Double(p1, p2));
+    for (int i = 1; i < points.length; i++) {
+      index[i] = index[i - 1] + Math.sqrt(points[i - 1].distance(points[i]));
```

Comment:
I was just reading this code line and was wondering if the Math.sqrt is needed as I thought we would just take the distance and maybe it is a left over from the manually distance calculation we had before. Please recheck.
Otherwise, what kind of name were you, @DanVanAtta, referring to? The index values are providing some kind of weights based on index value differences to the point.X and point.Y values for the interpolation.